File: C/BEGINNER/1047 Game Time with Minutes.c
#include <stdio.h>

int main()
{
    // start hour, start minute, end hour, end minute, result hour, result minute
    int startHour, startMinute, endHour, endMinute, resultHour, resultMinute;
    scanf("%d %d %d %d", &startHour, &startMinute, &endHour, &endMinute);
    
    // HOURS
    if (endHour > startHour)
    {
        if (endMinute >= startMinute)
            resultHour = (endHour - startHour);
        else if (startMinute > endMinute)
            resultHour = (endHour - startHour) - 1;
    }
    else if (startHour > endHour)
    {
        if (endMinute >= startMinute)
            resultHour = (24 - (startHour - endHour));
        else if (startMinute > endMinute)
            resultHour = (23 - (startHour - endHour));
    }
    else
    {
        if (endMinute > startMinute)
            resultHour = (endHour - startHour);
        else if (startMinute > endMinute)
            resultHour = (23 - (endHour - startHour));
        else
            resultHour = 24;
    }

    // MINUTES
    if (endMinute >= startMinute)
        resultMinute = (endMinute - startMinute);
    else
        resultMinute = (60 - (startMinute - endMinute));

    printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)\n", resultHour, resultMinute);
    return 0;
}

File: Python/BEGINNER/1047 Game Time with Minutes.py
# hi = start hour
# hf = final hour
# mi = start minute
# mf = final minute


hi, mi, hf, mf = map(int, input().split())

if hf > hi or mf > mi or hf < hi or mf < mi or hf == hi or mf == mi:
    if hf > hi and mf > mi:
        print(f'O JOGO DUROU {hf - hi} HORA(S) E {(mf - mi)} MINUTO(S)')

    elif hf > hi and mf < mi:
        print(f'O JOGO DUROU {(hf - hi) - 1 if (hi+1) != hf else 0} HORA(S) E {60 - (mi - mf)} MINUTO(S)')

    elif hf < hi and mf > mi:
        print(f'O JOGO DUROU {24 - (hi - hf)} HORA(S) E {(mf - mi)} MINUTO(S)')

    elif hf < hi and mf < mi:
        print(f'O JOGO DUROU {(24 - (hi - hf)) - 1 if (hf + 1) != hi else (24 - (hi - hf))} HORA(S) E {60 - (mi - mf)} MINUTO(S)')

    elif hf == hi and mf == mi:
        print(f'O JOGO DUROU {24 - (hf - hi)} HORA(S) E {0} MINUTO(S)')

    elif hf == hi and mf > mi:
        print(f'O JOGO DUROU {0} HORA(S) E {(mf - mi)} MINUTO(S)')

    elif hf == hi and mf < mi:
        print(f'O JOGO DUROU {24 - 1} HORA(S) E {60 - (mi - mf)} MINUTO(S)')

    elif hf > hi and mf == mi:
        print(f'O JOGO DUROU {(hf - hi) + 1} HORA(S) E {0} MINUTO(S)')

    else:
        print(f'O JOGO DUROU {24 - (hi - hf)} HORA(S) E {0} MINUTO(S)')
